src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1521,6 +1454,8 @@ public override BoundNode VisitObjectCreationExpression(BoundObjectCreationExpre
                             }
                         }
                     }
+
+                    this.State[slot] = resultState;
```

Comment:
>this.State[slot] = resultState; [](start = 20, length = 31)

Should this be pushed two lines lower even (outside the `if ((object)type != null)`)?